cmd/daemonset-check/cleanup.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// cleanUp triggers check clean up and waits for all rogue daemonsets to clear
+func cleanUp(ctx context.Context) error {
+
+	log.Debugln("Allowing clean up", checkTimeLimit, "to finish.")
+
+	// Clean up daemonsets and daemonset pods
+	// deleteDS not only issues a delete on the rogue daemonset but also on the rogue daemonset pods
+	log.Infoln("Cleaning up daemonsets and daemonset pods")
+
+	daemonSets, err := getAllDaemonsets()
+	if err != nil {
+		return err
+	}
+
+	// If any rogue daemonsets are found, proceed to remove them.
+	if len(daemonSets) > 0 {
+		for _, ds := range daemonSets {
+			log.Infoln("Removing rogue daemonset:", ds.Name)
+			err := remove(ds.Name, ctx)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	log.Infoln("Finished cleanup. No rogue daemonsets or daemonset pods exist")
+	return nil
+}
```

Comment:
instead of having a whole separate `runCheckCleanup` function like before, we just use the `remove` function to remove any daemonset created by the khcheck since there shouldn't be any left around. before there was a ton of logic to determine whether or not the daemonset was rogue but any daemonset left after a check run should be removed i believe

Reply:
this simplifies a LOT of the cleanUp stuff we were doing before